presto-main/src/main/java/io/prestosql/execution/SqlQueryExecution.java

```diff
@@ -352,6 +365,7 @@ public void start()
                 }
 
                 PlanRoot plan = planQuery();
+                registerDynamicFilteringQuery();
```

Comment:
Why do this before distributed planning?  In the old code this was performed after starting, which is a few lines down in this file.

Reply:
* `DynamicFilterService` needs plan to extract DF, so query registration has to happen after planning.
* dynamic filter suppliers are created in `planDistribution` just line below.

Registering query here removes the need to support `query has not been registered yet` case in `DynamicFilterService`, which simplifies https://github.com/prestosql/presto/pull/4224.

I will add comment to code